akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/BehaviorTestKitSpec.scala

```diff
@@ -395,6 +399,19 @@ class BehaviorTestKitSpec extends AnyWordSpec with Matchers with LogCapturing {
       val newChild = testkit.expectEffectType[Spawned[_]]
       child.childName shouldBe newChild.childName
     }
+
+    "start and stop child and then stop testkit itself" in {
+      val testkit = BehaviorTestKit(Parent.init)
+      testkit.run(SpawnChild)
+      val child = testkit.expectEffectType[Spawned[String]]
+      testkit.run(StopChild(child.ref))
+      testkit.expectEffect(Stopped(child.childName))
+
+      val ok = testkit.stopSelf(StopSelf)
+      assertEquals(true, ok)
```

Comment:
I'm not sure I understand the purpose of this feature. Couldn't this specific `StopSelf` message be tested with and ordinary `testkit.run(StopSelf)`?

The issue https://github.com/akka/akka/issues/29780 talks about the need to simulate that the parent stops a child actor, where the parent here would be the testkit itself. That would correspond to `context.stop(childRef)`. In that case I don't see why `def stopSelf(message)` should take a message parameter?

Reply:
Thank you for your reply. In fact, I'm still considering it because I'm not familiar with akka, and I'll update PR later.